src/nexmark/queries/q13.rs

```diff
@@ -0,0 +1,162 @@
+};
+
+use csv;
+
+/// Query 13: Bounded Side Input Join (Not in original suite)
+///
+/// Joins a stream to a bounded side input, modeling basic stream enrichment.
+///
+/// TODO: use the new "filesystem" connector once FLINK-17397 is done
+/// ```sql
+/// CREATE TABLE side_input (
+///   key BIGINT,
+///   `value` VARCHAR
+/// ) WITH (
+///   'connector.type' = 'filesystem',
+///   'connector.path' = 'file://${FLINK_HOME}/data/side_input.txt',
+///   'format.type' = 'csv'
+/// );
+///
+/// CREATE TABLE discard_sink (
+///   auction  BIGINT,
+///   bidder  BIGINT,
+///   price  BIGINT,
+///   dateTime  TIMESTAMP(3),
+///   `value`  VARCHAR
+/// ) WITH (
+///   'connector' = 'blackhole'
+/// );
+///
+/// INSERT INTO discard_sink
+/// SELECT
+///     B.auction,
+///     B.bidder,
+///     B.price,
+///     B.dateTime,
+///     S.`value`
+/// FROM (SELECT *, PROCTIME() as p_time FROM bid) B
+/// JOIN side_input FOR SYSTEM_TIME AS OF B.p_time AS S
+/// ON mod(B.auction, 10000) = S.key;
+/// ```
+///
+/// NOTE: although the Flink test uses a static file as the side input, the
+/// query itself allows joining the temporal table from the filesystem file that
+/// is updated while the query runs, joining the temporal table using process
+/// time. Flink itself ensures that the file is monitored for changes. The
+/// [current documentation for this connector](https://nightlies.apache.org/flink/flink-docs-master/docs/connectors/table/filesystem/)
+/// is not that used above, since the new connector (in 1.17-SNAPSHOT) does not
+/// allow specifying a file name, but only a directory to monitor. Rather, the
+/// Nexmark test appears to have used the previous legacy filesystem connector, [stable filesystem connector](https://nightlies.apache.org/flink/flink-docs-release-1.15/docs/connectors/datastream/filesystem/)
+/// which does allow specifying a file path.
+///
+/// Also see [Flink's Join with a Temporal Table](https://nightlies.apache.org/flink/flink-docs-release-1.11/dev/table/streaming/joins.html#join-with-a-temporal-table).
+///
+/// So, although Flink supports monitoring the side-loaded file for updates, a
+/// simple static file is used for this bounded side-input for the Nexmark tests
+/// and that is also what is tested here.
+
+const SIDE_INPUT_CSV: &str = "benches/nexmark/data/side_input.txt";
+
+type Q13Stream = Stream<Circuit<()>, OrdZSet<(u64, u64, usize, u64, String), isize>>;
+
+fn read_side_input<R: Read>(reader: R) -> Result<HashMap<usize, String>> {
+    let reader = BufReader::new(reader);
+    let mut csv_reader = csv::ReaderBuilder::new()
+        .has_headers(false)
+        .from_reader(reader);
+    let mut hm: HashMap<usize, String> = HashMap::new();
+    for result in csv_reader.deserialize() {
+        let (key, val): (usize, String) = result?;
+        hm.insert(key, val);
+    }
+    Ok(hm)
+}
+
+pub fn q13(input: NexmarkStream) -> Q13Stream {
+    let side_input = read_side_input(File::open(SIDE_INPUT_CSV).unwrap()).unwrap();
```

Comment:
We should feed `side_input` via the input handle API as a normal input stream and use the `join` operator to join with `input`. As you describe in the comment above, in general we should implement a file listener that ingests new records as they get appended to the file.  But in this case, since we know that the file is static, we can feed the entire contents as a separate transaction executed before starting the generator. 

Re processing time modeling: Processing time doesn't need to be a first-class primitive -- it's just a computed column added to a stream when it gets ingested by DBSP. One option would be to add it as a `p_time` field to `Bids`. The generator sets this field to zero.  Queries that use processing time set the value of the field, e.g., using `SystemTime::now()`.  The `p_time` field can then be used as any other field to filter/join on.